hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -690,6 +708,60 @@ BlobProperty getContainerProperty(TracingContext tracingContext) throws AzureBlo
     return blobProperty;
   }
 
+  /**
+   * Gets  User-defined properties(metadata) of the blob over blob endpoint
+   *
+   */
+  public Hashtable<String, String> getBlobMetadata(final Path path,
+      TracingContext tracingContext) throws AzureBlobFileSystemException {
+
+    final Hashtable<String, String> metadata;
+    final AbfsRestOperation op = client.getBlobMetadata(path, tracingContext);
+
+    metadata = parseResponseHeadersToHashTable(op.getResult());
+    return metadata;
+  }
+
+  public void setBlobMetadata(final Path path, final Hashtable<String, String> metadata,
+      TracingContext tracingContext) throws AzureBlobFileSystemException {
+    try (AbfsPerfInfo perfInfo = startTracking("setPathProperties", "setPathProperties")){
```

Comment:
Why is the debugger suggesting setPathProperties here ?

Reply:
Rectified